resources/js/project-report.js

```diff
@@ -0,0 +1,109 @@
+/*
+    This is what's included in project_diff_report.html
+    to make the violation table work.
+ */
+
+function makeCodeLink(violation) {
```

Comment:
I'd move these functions and consts down to the main function scope (Line 31), otherwise these are global vars...